src/main/java/com/example/gradecalculator/config/WebSecurityConfig.java

```diff
@@ -37,7 +38,10 @@ public UserDetailsService userDetailsService() {
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         http
-                .authorizeRequests((requests) -> requests.anyRequest().permitAll())
+                .authorizeHttpRequests((requests) -> requests.anyRequest().permitAll())
+                .csrf(csrf -> csrf
+                        .ignoringRequestMatchers(new AntPathRequestMatcher("/subject/delete/{ID}"))
```

Comment:
why is this necessary?

Reply:
This allows user to delete Subjects  not only on view layer also when he clicks on delete button subject gets removed in DB too

Reply:
But why is adding csrf and then ignoring the delete route necessary?
Can you explain why you decided to use CSRF?

Reply:
It was the only solution that I found for this error: Failed to load resource: the server responded with a status of 403 (). Which as I understood indicates permission issues. And disabling CSRF permits user to delete unwished subject from DB. Ignoring the delete route is necessary, it says not to use CSRF protection on /subject/delete/{ID} endpoint thence user can delete subject

Reply:
I didn't add csrf, I disabled. According to my research csrf protection is automatically enabled in Spring especially on DELETE MAPPING so I disabled it in order user will be  permitted to delete

Reply:
done